agent/consul/options.go

```diff
@@ -15,5 +16,9 @@ type Deps struct {
 	Tokens          *token.Store
 	Router          *router.Router
 	ConnPool        *pool.ConnPool
-	GRPCConnPool    *grpc.ClientConnPool
+	GRPCConnPool    GRPCClientConner
```

Comment:
This was done so that tests which use `Agent.New` but don't actually need a gRPC connection pool can use a no-op fake.